src/txmempool.cpp

```diff
@@ -429,5 +758,58 @@ bool CCoinsViewMemPool::HaveCoins(const uint256 &txid) const {
 
 size_t CTxMemPool::DynamicMemoryUsage() const {
     LOCK(cs);
-    return memusage::DynamicUsage(mapTx) + memusage::DynamicUsage(mapNextTx) + memusage::DynamicUsage(mapDeltas) + cachedInnerUsage;
+    // Estimate the overhead of mapTx to be 9 pointers + an allocation, as no exact formula for boost::multi_index_contained is implemented.
+    return memusage::MallocUsage(sizeof(CTxMemPoolEntry) + 9 * sizeof(void*)) * mapTx.size() + memusage::DynamicUsage(mapNextTx) + memusage::DynamicUsage(mapDeltas) + memusage::DynamicUsage(mapLinks) + cachedInnerUsage;
+}
+
+void CTxMemPool::RemoveStaged(setEntries &stage) {
```

Comment:
You can also AssertLockHeld(cs), otherwise, to make things more readable.

On September 16, 2015 11:05:34 AM CDT, Alex Morcos <notifications@github.com> wrote:
>> @@ -429,5 +758,58 @@ bool CCoinsViewMemPool::HaveCoins(const uint256
>&txid) const {
>>  
>>  size_t CTxMemPool::DynamicMemoryUsage() const {
>>      LOCK(cs);
>> -    return memusage::DynamicUsage(mapTx) +
>memusage::DynamicUsage(mapNextTx) + memusage::DynamicUsage(mapDeltas) +
>cachedInnerUsage;
>> +    // Estimate the overhead of mapTx to be 9 pointers + an
>allocation, as no exact formula for boost::multi_index_contained is
>implemented.
>> +    return memusage::MallocUsage(sizeof(CTxMemPoolEntry) + 9 *
>sizeof(void*)) * mapTx.size() + memusage::DynamicUsage(mapNextTx) +
>memusage::DynamicUsage(mapDeltas) + memusage::DynamicUsage(mapLinks) +
>cachedInnerUsage;
>> +}
>> +
>> +void CTxMemPool::RemoveStaged(setEntries &stage) {
>
>Needs LOCK(cs) if ever called from somewhere else.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/bitcoin/bitcoin/pull/6654/files#r39649424